app.rb

```diff
@@ -0,0 +1,137 @@
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_books_with_index
+    @books.each_with_index { |book, i| puts "#{i}) Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_people_with_index
+    @people.each_with_index do |person, i|
+      puts "#{i}) [#{person.class.name}] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    student_or_teacher = gets.chomp.to_i
+    case student_or_teacher
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts "Invalid choice. Please enter a valid option. (#{student_or_teacher})"
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp.to_s
+
+    print 'Has parent permission? [Y / N]: '
+    parent_permission = gets.chomp.to_s
+
+    if parent_permission =~ /^[Yy]/
+      student = Student.new('Unknown', age, name, parent_permission: true)
+    elsif parent_permission =~ /^[Nn]/
+      student = Student.new('Unknown', age, name, parent_permission: false)
+    else
+      puts "Invalid choice. Please enter a valid option. (#{parent_permission})"
+      return
+    end
+
+    @people.push(student)
+    puts 'Person created successfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp.to_s
+
+    print 'Specialization: '
+    specialization = gets.chomp.to_s
+
+    teacher = Teacher.new(specialization, age, name)
+    @people.push(teacher)
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp.to_s
+
+    print 'Author: '
+    author = gets.chomp.to_s
+
+    @books.push(Book.new(title, author))
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    list_books_with_index
+    book_index = gets.chomp.to_i
+    unless (0...@books.length).include?(book_index)
+      puts "Error adding a record. Book #{book_index} doesn't exist"
+      return
+    end
+    book = @books[book_index]
+    puts "\nSelect a person from the following list by number (not id)"
+    list_people_with_index
+    person_index = gets.chomp.to_i
+    unless (0...@people.length).include?(person_index)
+      puts "Error adding a record. Person #{person_index} doesn't exist"
+      return
+    end
+    person = @people[person_index]
+    print 'Date: '
+    date = gets.chomp.to_s
+    @rentals.push(Rental.new(date, book, person))
+    puts 'Rental created successfully'
+  end
```

Comment:
- [ ] An error occurs in the initialize method of the `Rental` class while trying to create a rental. Please see error below and try to fix it 👇 
![image](https://github.com/SaveryIV/OOP-School-Library/assets/46839548/e04ed7a3-660c-4f66-bb14-c012d9d7c37f)

Reply:
I cant solve it, i don't know what i can do